code/modules/mob/living/carbon/carbon.dm

```diff
@@ -21,13 +21,13 @@
 	if(!held_index)
 		held_index = (active_hand_index % held_items.len)+1
 
+	// Checks if the item in your hand is has the two_handed component
 	var/obj/item/item_in_hand = src.get_active_held_item()
-	if(item_in_hand) //this segment checks if the item in your hand is twohanded.
-		var/obj/item/twohanded/TH = item_in_hand
-		if(istype(TH))
-			if(TH.wielded == 1)
-				to_chat(usr, "<span class='warning'>Your other hand is too busy holding [TH].</span>")
-				return
+	if(item_in_hand)
+		var/datum/component/two_handed/comp_twohand = item_in_hand.GetComponent(/datum/component/two_handed)
+		if(comp_twohand && comp_twohand.wielded)

```

Comment:
Make a signal for swapping hands and interrupt it from the component

Reply:
The signal has been made and is used here instead.